eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -362,4 +321,72 @@ function  Get-LatestReleaseDateFromChangeLog
   $changeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $ChangeLogLocation
   $latestVersion = $changeLogEntries[0].ReleaseStatus.Trim("()")
   return ($latestVersion -as [DateTime])
+}
+
+function Confirm-LikeForRelease {
+  param (
+    [Parameter(Mandatory = $true)]
+    $changeLogEntry,
+    [Parameter(Mandatory = $true)]
+    $changeLogFileDateHistory
+  )
+
+  $isValid = $true
+  if ($changeLogEntry.ReleaseStatus -eq $CHANGELOG_UNRELEASED_STATUS) {
+    LogError "Entry has no release date set. Please ensure to set a release date with format '$CHANGELOG_DATE_FORMAT'. See https://aka.ms/azsdk/guideline/changelogs for more info."
+    $isValid = $false
+  }
+  else {
+    $status = $changeLogEntry.ReleaseStatus.Trim().Trim("()")
+    try {
+      $releaseDate = [DateTime]$status
+      if ($status -ne ($releaseDate.ToString($CHANGELOG_DATE_FORMAT)))
+      {
+        LogError "Date must be in the format $($CHANGELOG_DATE_FORMAT). See https://aka.ms/azsdk/guideline/changelogs for more info."
+        $isValid = $false
+      }
+
+      $dateHistory = $changeLogFileDateHistory | ForEach-Object { $_.Trim("()") } | Sort-Object -Descending
```

Comment:
We should probably use a common sort helper method so we don't have differences in how we sort here vs in the update.

Reply:
This one is sorting only the dates though. If we want to have the same sorting, then we can just use the order of the changelog entries which is almost always already sorted by prior processes.

Reply:
I think we should use the same sorting as the change log entries to just be sure they are consistent. Currently we sort the entries right before we save them so I just want to be sure that we are consistent.